Userland/Libraries/LibWeb/HTML/WindowOrWorkerGlobalScope.cpp

```diff
@@ -742,4 +743,81 @@ JS::NonnullGCPtr<JS::Object> WindowOrWorkerGlobalScopeMixin::supported_entry_typ
     return *m_supported_entry_types_array;
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#dom-reporterror
+void WindowOrWorkerGlobalScopeMixin::report_error(JS::Value e)
+{
+    auto& target = static_cast<DOM::EventTarget&>(this_impl());
+    auto& realm = relevant_realm(target);
+    auto script_or_module = realm.vm().get_active_script_or_module();
+
+    // FIXME: Get the current position in the script.
+    auto line = 0;
+    auto col = 0;
+
+    // 1. If target is in error reporting mode, then return; the error is not handled.
+    if (m_error_reporting_mode) {
+        report_exception_to_console(e, realm, ErrorInPromise::No);
+        return;
+    }
+
+    // 2. Let target be in error reporting mode.
+    m_error_reporting_mode = true;
+
+    // 3. Let message be an implementation-defined string describing the error in a helpful manner.
+    auto message = e.to_string_without_side_effects();
```

Comment:
Try this instead:

```suggestion
    String message;
    if (e.is_object()) {
        auto& object = e.as_object();
        message = object.get_without_side_effects(realm.vm().names.message).value_or(JS::js_undefined()).to_string_without_side_effects();
    } else {
        message = e.to_string_without_side_effects();
    }
```

Reply:
Kenneth beat me to it by a few seconds, here's what I had.
```c++
// 3. Let message be an implementation-defined string describing the error in a helpful manner.
    auto& vm = realm.vm();
    auto message = [&]() {
        if (e.is_object()) {
            auto& object = e.as_object();
            if (is<JS::Error>(object)) {
                auto const& error = static_cast<JS::Error const&>(object);
                if (MUST(error.has_own_property(vm.names.message))) {
                    auto message = MUST(error.get(vm.names.message));
                    return message.to_string_without_side_effects();
                }
            }
        }

        return MUST(String::formatted("Uncaught exception: {}", e.to_string_without_side_effects()));
    }();
```

Because the spec says that `message` is an implementation defined String, you have a lot of leeway over how exactly you format the error message.

Reply:
@kennethmyhra @tcl3 Thank you for the idea. Applied in [d146096](https://github.com/LadybirdBrowser/ladybird/pull/392/commits/d1460963f3f4ada8ce282708e45d6b3b5807bebd)